src/wp-includes/class-wp-theme-json.php

```diff
@@ -665,9 +665,17 @@ protected static function sanitize( $input, $valid_block_names, $valid_element_n
 		$schema_styles_blocks   = array();
 		$schema_settings_blocks = array();
 		foreach ( $valid_block_names as $block ) {
-			$schema_settings_blocks[ $block ]           = static::VALID_SETTINGS;
-			$schema_styles_blocks[ $block ]             = $styles_non_top_level;
-			$schema_styles_blocks[ $block ]['elements'] = $schema_styles_elements;
+			// Build the schema for each block style variation.
+			$style_variation_names    = isset( $input['styles']['blocks'][ $block ]['variations'] ) ? array_keys( $input['styles']['blocks'][ $block ]['variations'] ) : array();
```

Comment:
What if the array structure exists, but the `'variations'` are not an array? What would happen?

https://3v4l.org/kHpHT

* >= PHP 8.0
```
Fatal error: Uncaught TypeError: array_keys(): Argument #1 ($array) must be of type array, string given
```
* > PHP 8.0:
```
Warning: array_keys() expects parameter 1 to be array, string given
```

To avoid these, also check if it is an array.

I'd also suggest switching to `! empty()` instead of `isset()`. Why? To avoid doing `array_keys()` on an empty array.

```suggestion
			$style_variation_names = array();
			if (
				! empty( $input['styles']['blocks'][ $block ]['variations'] ) &&
				is_array( $input['styles']['blocks'][ $block ]['variations'] )
			) {
				$style_variation_names = array_keys( $input['styles']['blocks'][ $block ]['variations'] );
			}
```

Reply:
Resolved here https://github.com/WordPress/wordpress-develop/pull/3941/commits/86ac232dd9f489a221770a9cdb35b0b4b07a1c7d.